hathor/indexes/deps_index.py

```diff
@@ -104,6 +105,12 @@ class DepsIndex(ABC):
       them.
     """
 
+    def init_loop_step(self, tx: BaseTransaction) -> None:
+        tx_meta = tx.get_metadata()
+        if tx_meta.voided_by:
```

Comment:
Why do you skip voided transactions? What about soft voided ones?

Reply:
I'll have to review this one, I think I should let `add_tx` deal with the tx state.

Reply:
I'm not sure whether this index should have voided transactions or not. I'd say we should add soft-voided txs so we keep downloading all dependencies correctly.